packages/sanity/src/structure/panes/documentList/SheetListCell.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable i18next/no-literal-string */
+import {Select, TextInput} from '@sanity/ui'
+import {type CellContext} from '@tanstack/react-table'
+import {type FormEvent, useCallback, useEffect, useState} from 'react'
+import {type SanityDocument} from 'sanity'
+
+interface SheetListCellProps extends CellContext<SanityDocument, unknown> {
+  type: any
+}
+
+export function SheetListCell(props: SheetListCellProps) {
+  const initialValue = props.getValue() || ''
+  // We need to keep and update the state of the cell normally
+  const [value, setValue] = useState(initialValue)
+
+  // When the input is blurred, we'll call our table meta's updateData function
+  const handleOnBlur = () => {
+    //@ts-expect-error - wip.
```

Comment:
There's still a fair bit of work coming up next as to how changes triggers document updates, and also more work will be down to render the cell inputs, so I've kept this light touch for now